ext/php8/php8/engine_hooks.c

```diff
@@ -758,10 +760,29 @@ zend_observer_fcall_handlers ddtrace_observer_fcall_init(zend_execute_data *exec
     return (zend_observer_fcall_handlers){NULL, NULL};
 }
 
-/* If the profiler is loaded, then this weak symbol will be overwritten by the
- * strong symbol provided by profiler.
+static void (*profiling_interrupt_function)(zend_execute_data *) = NULL;
+
+/**
+ * The message handler is used to determine if the profiler is loaded, and if
+ * so it will locate certain symbols so cross-product features can be enabled.
  */
-__attribute__((weak)) extern void datadog_profiling_interrupt_function(zend_execute_data *execute_data);
+void ddtrace_message_handler(int message, void *arg) {
+    if (UNEXPECTED(message != ZEND_EXTMSG_NEW_EXTENSION)) {
+        // There are currently no other defined messages.
+        return;
+    }
+
+    zend_extension *extension = (zend_extension *)arg;
+    if (extension->name && strcmp(extension->name, "datadog-profiling") == 0) {
+        DL_HANDLE handle = extension->handle;
+
+        profiling_interrupt_function = DL_FETCH_SYMBOL(handle, "datadog_profiling_interrupt_function");
```

Comment:
I think this is good--does the behavior of `DL_FETCH_SYMBOL()` around `DLSYM_NEEDS_UNDERSCORE` matter for you?

Reply:
I wondered about the underscore prefix as well, but I assume given the way the symbols are exported on the profiler (`datadog-profiling.sym`) it never will be mangled.